File: app/src/main/java/com/example/Sleepy/main/modules/wake/domain/WakeCards.java
package com.example.Sleepy.main.modules.wake.domain;

import java.util.Calendar;
import java.util.Date;

public class WakeCards {
    private String Time;
    private String RemainingTime;
    private Calendar TriggerTime = Calendar.getInstance();

    public WakeCards(String formatTime, String remainingTimeMinute, Calendar triggerTime) {
        Time = formatTime;
        RemainingTime = remainingTimeMinute;
        TriggerTime.set(triggerTime.get(Calendar.YEAR), triggerTime.get(Calendar.MONTH), triggerTime.get(Calendar.DATE), triggerTime.get(Calendar.HOUR_OF_DAY), triggerTime.get(Calendar.MINUTE));
    }

    public String getRemainingTime() {
        return RemainingTime;
    }

    public void setRemainingTime(String remainingTime) {
        RemainingTime = remainingTime;
    }

    public String getTime() {
        return Time;
    }

    public void setTime(String time) {
        Time = time;
    }

    public void setTriggerTime(Calendar triggerTime) {
        TriggerTime.setTime(triggerTime.getTime());
    }

    public Calendar getTriggerTime() {
        return TriggerTime;
    }
}

File: app/src/main/java/com/example/sleepy/presentation/wake/recycler/WakeCards.kt
package com.example.sleepy.presentation.wake.recycler

import java.util.*

class WakeCards(var time: String, var remainingTime: String, triggerTime: Calendar) {
    private val triggerTime = Calendar.getInstance()

    fun setTriggerTime(triggerTime: Calendar) {
        this.triggerTime.time = triggerTime.time
    }

    fun getTriggerTime(): Calendar {
        return triggerTime
    }

    init {
        this.triggerTime[triggerTime[Calendar.YEAR], triggerTime[Calendar.MONTH], triggerTime[Calendar.DATE], triggerTime[Calendar.HOUR_OF_DAY]] =
            triggerTime[Calendar.MINUTE]
    }
}